src/Arcus.Messaging.Pumps.ServiceBus/Extensions/IServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,175 @@
+﻿using System;
+using System.Threading.Tasks;
+using Arcus.Messaging.Pumps.ServiceBus;
+using Arcus.Security.Core;
+using GuardNet;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Hosting;
+
+// ReSharper disable once CheckNamespace
+namespace Microsoft.Extensions.DependencyInjection
+{
+    // ReSharper disable once InconsistentNaming²
+    public static class IServiceCollectionExtensions
```

Comment:
Good job on trying to provide all possible, usable combinations for registrations!

Reply:
My pleasure, good user experience is my top 1 priority which is what we're trying to leverage here.